docs/src/content/docs/guides/stages.mdx

```diff
@@ -0,0 +1,111 @@
+| voice               | Yes                    |
+| languageHint        | Yes                    |
+| initialMessages     | Yes                    |
+| selectedTools       | Yes                    |
+| firstSpeaker        | Yes                    |
+| model               | No                     |
+| joinTimeout         | No                     |
+| maxDuration         | No                     |
+| timeExceededMessage | No                     |
+| inactivityMessages  | No                     |
+| medium              | No                     |
+| initiator           | No                     |
+| recordingEnabled    | No                     |
+
+The response body must contain the new values for any properties you want to change in the new stage.
+
+#### 3. Set Up Stage Transitions
+Prompt the agent to use tool calls to initiate stage changes at appropriate points in the conversation.
+
+#### 4. Configure New Stage Parameters
+When transitioning to a new stage, provide the updated parameters (e.g., system prompt) in the tool call.
+
+### Example Stage Change Implementation
+
+Here's a basic example of how to implement a new call stage.
+
+First, we create a tool that is responsible for changing stages:
+
+import callStageTool from '../../codesnippets/call-stage.txt?raw';
+export const title = "Basic stage change tool"
+
+<Code code={callStageTool} lang="js" title={title} />
+
+We also need to ensure that we have instructed our agent to use the tool and that we add the tool to our `selectedTools` during the creation of the call.
+
+import callStage from '../../codesnippets/call-stage-prompt.txt?raw';
+export const cstitle = "Updating our systemPrompt and selectedTools"
+
+<Code code={callStage} lang="js" title={cstitle} />
+
+
+:::tip[Inheritance]
+New stages inherit all properties from the previous stage. You can selectively overwrite properties as needed when defining a new stage. See the [above table](#2-implement-a-stage-change-tool) for more.
+:::
+
+## Use Cases for Call Stages
+
+Call Stages are particularly useful for complex conversational flows. Here are some example scenarios:
+
+**Data Gathering** → Scenarios where the agent needs to collect a lot of data. Examples: job applications, medical intake forms, applying for a mortgage.
+
+Here are potential stages for a **Mortgage Application**:
+- Stage 1: Greeting and basic information gathering
+- Stage 2: Financial assessment
+- Stage 3: Property evaluation
+- Stage 4: Presentation of loan options
+- Stage 5: Hand-off to loan officer
+
+**Switching Contexts** → Scenarios where the agent needs to navigate different contexts. Examples: customer support escalation, triaging IT issues.
+    
+Let's consider what the potential stages might be for **Customer Support**: 
+- Stage 1: Initial greeting and problem identification
+- Stage 2: Troubleshooting
+- Stage 3: Resolution or escalation (to another stage or to a human support agent)
+
+## Conclusion
```

Comment:
Probably worth adding a section about how call stages impact other rest endpoints, namely that getting a call or its tools/messages will reflect the call's last stage as you'd expect if you don't use stage changes. For anyone knowingly using stages, the stage-centric APIs are likely to be preferable.

Reply:
good call